src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ComparerWithState.cs

```diff
@@ -2,18 +2,36 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
+
 using System;
 using System.Collections.Immutable;
 
 namespace Roslyn.Utilities
 {
     static internal class ComparerWithState
     {
-        public static int CompareTo<T, S>(T first, T second, S state, ImmutableArray<Func<T, S, IComparable>> comparableMethods)
+        public static int CompareTo<T, S>(T first, T second, S state, ImmutableArray<Func<T, S, IComparable?>> comparableMethods)
```

Comment:
Did the previous users of this only ever return non-null for the IComparable being returned? Looking at some of the uses it's almost always a boxed (ugh!) bool or int or something. There's a couple strings but those are non-null too. If nothing else, it's not clear to me why all callers unilaterally want null handled in the way it's being handled, since I could imagine some want null to mean first and some to last.

Put another way: this seems to be technically correct, but do the callers actually want this, or is just throwing on null more appropriate?

Reply:
(it's possible some caller does want this too and I just missed it, but at least a few minute glance of callers didn't find any....)

Reply:
I'm also surprised this doesn't return `IComparable<T>` to avoid boxing.

Reply:
When the caller was updated to support null inputs, it cascaded to this.

Reply:
I guess my thesis is since the old code clearly didn't handle nulls, so either no nulls were getting through, or they could have gotten through. If the former is the case, it seems we need a better annotation somewhere else. If the latter is the case, is this the right fix for those features? Which features were they? Basically, why is the null sorting here the correct behavior for those features?

Reply:
Or perhaps another question: if nulls can get here, what unit tests do we need to add to those features to cover this?

Reply:
Compiling against the netcoreapp3.1 reference assemblies means we are compiling against annotated framework code for the first time. The annotations in that library were inconsistent with our usage, so I have updated our implementations to no longer have OHI warnings in the new build.

Reply:
Which change was it?

Reply:
I think I found a simpler approach now